lib/rack/handler/puma.rb

```diff
@@ -1,114 +1,118 @@
 # frozen_string_literal: true
 
-require 'rack/handler'
-
-module Rack
-  module Handler
-    module Puma
-      DEFAULT_OPTIONS = {
-        :Verbose => false,
-        :Silent  => false
-      }
-
-      def self.config(app, options = {})
-        require_relative '../../puma'
-        require_relative '../../puma/configuration'
-        require_relative '../../puma/log_writer'
-        require_relative '../../puma/launcher'
-
-        default_options = DEFAULT_OPTIONS.dup
-
-        # Libraries pass in values such as :Port and there is no way to determine
-        # if it is a default provided by the library or a special value provided
-        # by the user. A special key `user_supplied_options` can be passed. This
-        # contains an array of all explicitly defined user options. We then
-        # know that all other values are defaults
-        if user_supplied_options = options.delete(:user_supplied_options)
-          (options.keys - user_supplied_options).each do |k|
-            default_options[k] = options.delete(k)
-          end
+# Code below is 'odd', but don't want to define a top level constant
+
+if Object.const_defined? :Rackup
+  require 'rackup/handler'
+  ::Rackup::Handler
+else
+  require 'rack/handler'
+  ::Rack::Handler
+end.module_eval <<'HEREDOC'
```

Comment:
To get rid of this `module_eval` we could create a new lib/rackup/handler.rb file that is required here. The file could either duplicate the code, or we are putting the code inside a module and are including this new module inside the new rackup/handler and rack/handler :thinking:

Reply:
The cleanest way is with `modulfe_eval`, but that also 'hides' the code from RuboCop, etc.  I changed the code to use include.  See the last commit.